engines/python/setup/djl_python/huggingface.py

```diff
@@ -496,41 +496,96 @@ def register_adapter(inputs: Input):
     """
     Registers lora adapter with the model.
     """
+    if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
+        raise NotImplementedError(
+            "LoRA adapter API is only supported for rolling batch.")
+
     adapter_name = inputs.get_property("name")
     adapter_path = inputs.get_property("src")
+    adapter_pin = inputs.get_property("pin") == "true"
     if not os.path.exists(adapter_path):
         raise ValueError(
             f"Only local LoRA models are supported. {adapter_path} is not a valid path"
         )
     logging.info(f"Registering adapter {adapter_name} from {adapter_path}")
     _service.adapter_registry[adapter_name] = inputs
+
+    result = _service.rolling_batch.add_lora(adapter_name, adapter_path)
+    if adapter_pin:
+        try:
+            if not result:
```

Comment:
this should be moved before the if adapter_pin check

Reply:
Done